File: src/test/java/lechuck/intellij/util/RegexUtilTest.java
package lechuck.intellij.util;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

import java.util.List;

public class RegexUtilTest {

    @Test
    public void testSplitBySpacePreservingQuotes() {
        assertEquals(List.of(), RegexUtil.splitBySpacePreservingQuotes(""));
        assertEquals(List.of("a", "b"), RegexUtil.splitBySpacePreservingQuotes("a b"));
        assertEquals(List.of("a", "a b"), RegexUtil.splitBySpacePreservingQuotes("a  \"a b\""));
    }
}

File: src/test/kotlin/lechuck/intellij/util/RegexUtilTest.kt
package lechuck.intellij.util

import lechuck.intellij.util.RegexUtil.splitBySpace
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class RegexUtilTest {
    @Test
    fun testSplitBySpace() {
        assertEquals(emptyList<String>(), splitBySpace(null))
        assertEquals(emptyList<String>(), splitBySpace(""))
        assertEquals(listOf("a", "b"), splitBySpace("a b"))
        assertEquals(listOf("a", "a b"), splitBySpace("""a  "a b" """))
    }
}